projects/fsastorefront-e2e-cypress/cypress/integration/common/cart-retrieve-quote-not-same-session.e2e-spec.ts

```diff
@@ -0,0 +1,104 @@
+      auto.populateAutoAnnuallyBMW();
+      auto.populateMainDriverInfo();
+      cy.get('[name=noOfDrivers]').select('0');
+      checkout.clickContinueButton();
+      autoIntegration.selectAutoSilver();
+      auto.addOptionalProductsSilver();
+      checkout.clickContinueButton();
+      auto.populatePersonalDetails();
+      auto.populateVehicleDetails();
+      auto.populateMainDriverData();
+      checkout.clickContinueButton();
+    });
+
+    it('Should logout / login and change main product ', () => {
+      cy.get('.SiteLogo').should('be.visible').click();
+      register.logout();
+      checkPageElements();
+      cy.get('cx-login').click();
+      register.login(registrationUser.email, registrationUser.password);
+      checkMyQuotesPage();
+      retrieveQuote('1', 'Auto Insurance');
+      //TODO: add changing main product once bug is resolved
+      checkout.checkCheckoutStep('Your Auto Insurance', '7');
+      checkout.checkProgressBarInsurance();
+      checkout.clickContinueButton();
+    });
+
+    it('Should bind auto quote', () => {
+      checkout.checkPersonalDetailsPage();
+      auto.populateVehicleDetails();
+      auto.populateMainDriverData();
+      checkout.clickContinueButton();
+      checkout.checkCheckoutStep('Your Auto Insurance', '7');
+      checkout.checkProgressBarInsurance();
+      checkout.checkAccordions('generalQuoteAccordions');
+      checkout.clickContinueButton();
+      checkout.ConfirmBindQuote();
+      checkout.checkAccordions('generalQuoteAccordions');
+      cy.get('.SiteLogo').should('be.visible').click();
+    });
+
+    it('Should start Savings checkout', () => {
+      checkout.startInsuranceCheckout('Savings');
+      checkout.checkCheckoutStep('Your Savings Insurance', '7');
+      checkout.checkProgressBarInsurance();
+      checkout.checkFirstCheckoutStep('Coverage');
+      savings.populateCoverageInformation();
+      checkout.clickContinueButton();
+      checkout.checkCheckoutStep('Your Savings Insurance', '7');
+      checkout.checkProgressBarInsurance();
+      savings.checkComparisonPage();
+      savings.checkSavingsComparisonTable();
+      savings.selecBalancedDeal();
+      checkout.checkCheckoutStep('Your Savings Insurance', '7');
+      savings.checkOptionalProductsBalancedDeal();
+      savings.checkMiniCart();
+      checkout.clickContinueButton();
+    });
+
+    it('Should logout / login and retrieve quotes ', () => {
+      cy.get('.SiteLogo').should('be.visible').click();
+      register.logout();
+      checkPageElements();
+      cy.get('cx-login').click();
+      register.login(registrationUser.email, registrationUser.password);
+      checkMyQuotesPage();
+      retrieveQuote('2', 'Savings Insurance');
+      cy.get('h2').should('contain.text', 'Savings Insurance');
+      checkout.checkCheckoutStep('Your Savings Insurance', '7');
+      checkout.checkProgressBarInsurance();
+      cy.get('.section-header-heading').should('contains.text', 'Add Options');
```

Comment:
shouldn't this be `contain.text` instead of `constainS.text`. How does it work anyway? Or are they both valid methods?

Reply:
both are valid, but for future i will change only to containS

Reply:
I wouldn't do that as a rule. Because as far as I could understand there are situations when it's better to use one over the other as described [here](https://stackoverflow.com/questions/50042670/cypress-test-is-contains-equivalent-to-shouldcontain). So based on that I would say that in this case `should(contain)` is a preferred way, since you are not chaining anything afterwards. Also it appears that `containS` is not intended to be used in this way (with `should`).